### PR TITLE
Expand rst_epilog

### PR DESCRIPTION
**Pull request by tcmetzger**
Following up on the discussion in https://github.com/bokeh/bokeh/pull/11129#discussion_r608312454, this PR expands Sphinx' `rst_epilog` for common substitutions.
This is a work in progress, I will keep adding more substitutions over the next few days.
cc @bryevdv

### CODE REVIEW

**Comment by tcmetzger** [comment]
BTW: Several files link to the standard Python webbrowser module (as mentioned in https://github.com/bokeh/bokeh/pull/11129#discussion_r608312454). Instead of using a substitution to link to the Python docs, I have used intersphinx to link there (commit 94fa0f8).

File: bokeh/client/session.py [hunk 1]
```diff
@@ -194,7 +194,8 @@ def show_session(session_id=None, url='default', session=None, browser=None, new
             browser (str, optional) : browser to show with (default: None)
                 For systems that support it, the **browser** argument allows
                 specifying which browser to display in, e.g. "safari", "firefox",
-                "opera", "windows-default" (see the ``webbrowser`` module
+                "opera", "windows-default" (see the
+                :doc:`webbrowser <python:library/webbrowser>` module
```
Comment (bryevdv) [inline bokeh/client/session.py:198]: Just speaking personally, if the choice is between a slightly longer (than 80chars) line, or a weird extra-short line, I find the longer line more readable.

**Comment by tcmetzger** [comment]
Also fixes #11215

**Comment by bryevdv** [comment]
@tcmetzger FYI there are now some conflicts to resolve due to the refactoring of some code in `bokeh.plotting` We might want to consider merging this as soon as those are resolved and splitting futher work in to future PRs (there's no need to collect everything in one PR)

**Comment by tcmetzger** [comment]
@bryevdv I have merged branch-2.4 and done some cleanup so that this branch could be merged. I'm planning on adding more substitutions in follow-up PRs.
I do not know why the macos-latest, 3.7 unit test fails - all the other tests seem to run fine.

**Comment by mattpap** [comment]
> I do not know why the macos-latest, 3.7 unit test fails - all the other tests seem to run fine.

Tests on windows and macos sometimes fail for no good reason. Usually this goes away after a rerun. However, upon a review of those failures, it's fine to just ignore them, especially if otherwise it would lead to wasting our CI testing capacity (one can't just rerun one unit test job on github actions).

**Pull request merged**